discovery_engine_core/core/src/storage/sqlite.rs

```diff
@@ -84,113 +92,301 @@ impl Storage for SqliteStorage {
+        Ok(())
+    }
+
+    async fn fetch_feed(&self) -> Result<Vec<ApiDocumentView>, Self::FeedScopeError> {
+        #[derive(sqlx::FromRow)]
+        struct _ApiDocumentView {
+            document: Uuid,
+            title: String,
+            snippet: String,
+            topic: String,
+            url: String,
+            image: Option<String>,
+            date_published: NaiveDateTime,
+            source: String,
+            market: String,
+            domain_rank: u32,
+            score: f32,
+            user_reaction: Option<u32>,
+            in_batch_index: u32,
+        }
+
+        let mut con = self.pool.acquire().await.map_err(DatabaseError::Sql)?;
+        sqlx::query_as::<_, _ApiDocumentView>(
+            "SELECT nr.document, nr.title, nr.snippet, nr.topic, nr.url, nr.image,
+            nr.datePublished, nr.source, nr.market, nc.domainRank, nc.score, ur.userReaction,
+            po.inBatchIndex
+            FROM NewsResource as nr, NewscatcherData as nc, UserReaction as ur,
+            FeedDocuments as fd, PresentationOrdering as po
+            ON fd.document = nr.document, fd.document = nc.document,
+            fd.document = ur.document, fd.document = po.document
+            ORDERED BY po.timestamp, po.inBatchIndex ASC;",
+        )
+        .fetch_all(&mut con)
+        .await
+        .map_err(DatabaseError::Sql)?
+        .into_iter()
+        .map(|doc| {
+            let url = Url::parse(&doc.url).map_err(|e| DatabaseError::TypeConversion(e.into()))?;
+            let image = doc
+                .image
+                .map(|url| Url::parse(&url).map_err(|e| DatabaseError::TypeConversion(e.into())))
+                .transpose()?;
+            let (lang_code, country_code) = doc.market.split_at(2);
+            let market = Market {
+                lang_code: lang_code.to_string(),
+                country_code: country_code.to_string(),
+            };
+
+            let news_resource = NewsResource {
+                title: doc.title,
+                snippet: doc.snippet,
+                topic: doc.topic,
+                url,
+                image,
+                date_published: doc.date_published,
+                source: doc.source,
+                market,
+            };
+            let newscatcher_data = NewscatcherData {
+                domain_rank: doc.domain_rank as usize,
+                score: Some(doc.score),
+            };
+            let user_reacted = doc.user_reaction.and_then(FromPrimitive::from_u32);
+
+            Ok(ApiDocumentView {
+                document: document::Id::from(doc.document),
+                news_resource,
+                newscatcher_data,
+                user_reacted,
+                in_batch_index: doc.in_batch_index as usize,
             })
-            .collect()
+        })
+        .collect()
+    }
+
+    async fn store_documents(
+        &self,
+        documents: &[document::Document],
+    ) -> Result<(), Self::FeedScopeError> {
```

Comment:
We probably should have a sqlite specific helper function which given a transaction and a list of documents will  insert them into `Documents`, `NewsResource` and `NewscatcherData`. So that we then can use it here and in the `ActiveDocumentScope`.

Reply:
moved it into a helper function